plugins/amd-kria/fu-amd-kria-device.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 Richard Hughes <richard@hughsie.com>
+ * Copyright 2024 Advanced Micro Devices Inc.
+ *
+ * This file is provided under a dual MIT/LGPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ * AMD Chooses the MIT license part of Dual MIT/LGPLv2 license agreement.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-amd-kria-device.h"
+
+struct _FuAmdKriaDevice {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAmdKriaDevice, fu_amd_kria_device, FU_TYPE_DEVICE)
+
+static gboolean
+fu_amd_kria_device_write_firmware(FuDevice *device,
+				  FuFirmware *firmware,
+				  FuProgress *progress,
+				  FwupdInstallFlags flags,
+				  GError **error)
+{
+	FuContext *ctx = fu_device_get_context(device);
+	g_autofree gchar *cod_path = NULL;
+	g_autoptr(GBytes) fw = NULL;
+
+	fw = fu_firmware_get_bytes(firmware, error);
+	if (fw == NULL)
+		return FALSE;
+	cod_path = g_build_filename(fu_context_get_esp_location(ctx),
+				    "EFI", "UpdateCapsule", "fwupd.cap", NULL);
+	g_print("using %s for capsule", cod_path);
+	if (!fu_path_mkdir_parent(cod_path, error))
+		return FALSE;
+	if (!fu_bytes_set_contents(cod_path, fw, error))
+		return FALSE;
+
+	return TRUE;
+}
+
+static void
+fu_amd_kria_device_init(FuAmdKriaDevice *self)
+{
+	fu_device_set_name(FU_DEVICE(self), "System Firmware");
+	fu_device_set_vendor(FU_DEVICE(self), "AMD");
```

Comment:
don't you need a vendor_id here too?

Reply:
It's set from the fu-amd-kria-plugin.c taken from the HWID.